google/cloud/internal/debug_string.h

```diff
@@ -99,6 +99,20 @@ class DebugFormatter {
     return *this;
   }
 
+  template <typename T>
+  DebugFormatter& Field(absl::string_view field_name,
+                        std::map<std::string, T> const& value) {
+    for (auto const& e : value) {
+      absl::StrAppend(&str_, Sep(), field_name, " {");
+      ++indent_;
+      absl::StrAppend(&str_, Sep(), "key: ", "\"", e.first, "\"");
+      absl::StrAppend(&str_, e.second.DebugString("value", options_, indent_));
```

Comment:
Food for thought: this only works if `T` has a `DebugString()` member function.  You already have a special case for `std::map<std::string, std::string>`, but what about `std::map<std::string, int>`?

Maybe there is a way to implement this with some ADL, such that we can add types over time.